server/api/student/student.controller.js

```diff
@@ -147,33 +143,37 @@ exports.outreachSummary = function(req, res) {
 exports.interventionSummary = function(req, res) {
   var pipeline = [{
     $group: {
-      _id: {student: '$student', school: '$school' },
-      records: { $push: '$$ROOT' }
-    } 
+      _id: {student: '$student'},
+      records: {$push: '$$ROOT'}
+    }
   }, {
     $project: {
       _id: 0,
       student: '$_id.student',
-      school: '$_id.school',
       // Might need to exclude fields if implemented
       records: 1
     }
   }];
-  if(req.user.role === 'teacher') {
-    pipeline.unshift({ 
-      $match: { school: req.user.assignment } 
+  if (req.user.role === 'teacher') {
+    pipeline.unshift({
+      $match: {school: req.user.assignment}
     });
   }
   Intervention.aggregate(pipeline, function(err, results) {
     if (err) handleError(res, err);
-    Student.populate(results, {path: 'student'})
-      .then(function(withStudent) {
-        School.populate(withStudent, 
-          {path: 'school'})
-            .then(function(final) {
-              return res.status(200).json(final);
-            });
-      });
+    Intervention.populate(results, {
```

Comment:
The aggregation is on Intervention. Populate should use Intervention as well even though you are populating a school. In the case you aren't only populating a student from the intervention aggregation, e.g. a student and a school, using Student to populate doesn't really make sense. It also makes it so you don't need to require the Student model in this file if it isn't used.